lib/pbench/test/functional/agent/cli/commands/tools/test_tools_list.py

```diff
@@ -1,31 +1,167 @@
+        assert exitcode == 0
+
+    def test_no_args(self, tool, agent_config):
+        command = ["pbench-list-tools"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        assert b"default: testhost.example.com ['perf']\n" == out
+        assert exitcode == 0
+
+    def test_group_existing(self, tool, agent_config):
+        command = ["pbench-list-tools", "--group", "default"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        assert b"default: testhost.example.com ['perf']\n" == out
+        assert exitcode == 0
+
+    def test_name_existing(self, tool, agent_config):
+        command = ["pbench-list-tools", "-n", "perf"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        assert b"tool name: perf groups: default\n" == out
+        assert exitcode == 0
+
+    def test_non_existent_group(self, tool, agent_config):
+        command = ["pbench-list-tools", "--group", "unknown"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        assert exitcode == 1
+
+    def test_non_existent_name(self, tool, agent_config):
+        command = ["pbench-list-tools", "-n", "unknown"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        # XXX 0.69.9 backward compatibility: non-existent names do not return errors
+        assert exitcode == 0
+
+    def test_existing_group_name(self, tool, agent_config):
+        command = ["pbench-list-tools", "--group", "default", "--name", "perf"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        assert b"tool name: perf groups: default\n" == out
+        assert exitcode == 0
+
+    def test_existing_group_non_existent_name(self, tool, agent_config):
+        command = ["pbench-list-tools", "--group", "default", "--name", "unknown"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        assert exitcode == 0
+
+    def test_non_existent_group_existing_name(self, tool, agent_config):
+        command = ["pbench-list-tools", "--group", "unknown", "--name", "perf"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        assert exitcode == 1
+
+
+class Test_list_tool_tools_registered_with_options:
+    @pytest.fixture
+    def tool(self, pbench_run):
+        for group in ["default", "test"]:
+            p = pbench_run / f"tools-v1-{group}" / "testhost.example.com"
+            p.mkdir(parents=True)
+            tool = p / "iostat"
+            tool.write_text("--interval=30")
+            tool = p / "mpstat"
+            tool.write_text("--interval=300")
+        return
+
+    def test_existing_group_options(self, tool, agent_config):
+        command = ["pbench-list-tools", "--group", "default", "--with-option"]
+        out, err, exitcode = pytest.helpers.capture(command)
+        # XXX this is the current output - pretty ugly
+        assert (
+            b"default: testhost.example.com [('iostat', '--interval=30'), ('mpstat', '--interval=300')]\n"
+            == out
+        )
```

Comment:
Yeah, we should be able to do better than that -- at least get the Pythonisms out of there and use a more human-readable format.  E.g.,
```
default: testhost.example.com:  iostat --interval=30; mpstat --interval=300
```
(I don't like the colon-colon, but I can't come up with anything that I like better.)

Reply:
Gone back to the 0.69.9 format for these (except that the hostname is added here): ugly, but no pythonisms.